### PR TITLE
Case sensitive

### LINKED ISSUE #135: Some console commands are case sensitive to player names
Both `give safwat dirt` and `give Safwat dirt` work.
However, `clear safwat` does not work while `clear Safwat` works.

### PR DESCRIPTION
**Pull request by bearbin**
Made the commands case-insensitive and also autocomplete names. Fixes #135

### CODE REVIEW

File: clear.lua [hunk 1]
```diff
@@ -9,10 +9,8 @@ function HandleClearCommand(Split, Player)
     if Player:HasPermission("core.admin.clear") then
         local InventoryCleared = false;
         local ClearInventory = function(OtherPlayer)
-            if (OtherPlayer:GetName() == Split[2]) then
-                OtherPlayer:GetInventory():Clear()
-                InventoryCleared = true
-            end
+            OtherPlayer:GetInventory():Clear()
+            InventoryCleared = true
```
Comment (madmaxoft) [inline clear.lua]: This is not so good. The `FindAndDoWithPlayer` function actually calls the callback for the nearest name, it doesn't have to match.
Consider there are three players, Admin, Xoft and Xof. Now admin wants to cleat Xof's inventory, so he types `/clear Xof`. Just before sending the command, Xof disconnects, and Xoft's inventory gets cleared.

**Comment by madmaxoft** [comment]
Question: Do we want case-insensitive, or case-sensitive names?

**Comment by bearbin** [comment]
We want case insensitive names.

File: clear.lua (hunk 1 above)
Comment (bearbin) [inline clear.lua]: But the likelihood of that happening is very small, and the usability benefit of being able to use partial names IMO outweights the possibility of doing the command to the wrong person.

**Comment by tigerw** [comment]
That was answered before, but please to consider reconsidering: #761

Seeing as bearing himself voted no, opinions can change! :P

**Comment by bearbin** [comment]
@tigerw That's a different issue - the actual command. This is discussing the player names in the command. I still think that the command names themselves should probably be case-sensitive, in order that plugins can make up their own mind as to what to do with different capitalisations of commands.

File: clear.lua (hunk 1 above)
Comment (SafwatHalaby) [inline clear.lua]: Perhaps show a confirmation message for critical commands like banning or clearing inventory?
"are you sure? Press Y to confirm and delete the Inventory of <Full player name here>"

Comment (bearbin) [inline clear.lua]: Better would be a delay where you could type `/clear cancel` to cancel the clearing. But IDK if that's the best solution.

**Pull request closed without merging**